poo/diagramas/proyectoHeladeria/index.ts

```diff
@@ -0,0 +1,124 @@
+class Helado{
+    private _codigo: number;
+    protected azucar: boolean;
+    private sabor: string;
+    private base: string;
+
+    public constructor(_codigo:number,_sabor:string,_base:string){
+        this._codigo=_codigo;
+        this.azucar= true;
+        this.sabor=_sabor;
+        this.base=_base;
+    }
+}
+
+class TortaHelada extends Helado{
+    protected vizcocho: boolean;
+    private conberturaChocolate: boolean;
+
+    public constructor(_codigo:number,_sabor:string,_base:string){
+        super(_codigo,_sabor,_base)
+        this.vizcocho= true;
+        this.conberturaChocolate= true;
+    }
+
+}
+
+class Sundae extends Helado{
+    private salsa: string;
+    private topping: string;
+
+    public constructor(_codigo:number,_sabor:string,_base:string,_salsa:string,_topping:string){
+        super(_codigo,_sabor,_base)
+        this.salsa=_salsa;
+        this.topping=_topping;
+    }
+}
+
+class Almendrado extends Helado{
+    protected almendras:boolean;
+
+    public constructor(_codigo:number,_sabor:string,_base:string){
+        super(_codigo,_sabor,_base)
+        this.almendras= true;
+    }
+}
+class Stock{
+    private _codigo: number;
+    private _precio: number;
+    private _cantidad: number;
+        
+    public constructor(_codigo:number,_precio:number,_cantidad:number){
+       this._codigo=_codigo;
+       this._precio=_precio;
+       (_cantidad==null||_cantidad==undefined)?this._cantidad=_cantidad=0:this._cantidad=_cantidad;
```

Comment:
Esta linea esta bien? this._cantidad=_cantidad=0

Reply:
NO, JAJAJAJA! Era: this._cantidad=0 : this._cantidad=_cantidad